src/metabase/fingerprinting.clj

```diff
@@ -194,24 +202,29 @@
       [col (fingerprint-field field)])))
 
 (defmethod fingerprint (class Table)
-  [table]
+  [opts table]
   (fingerprint-query (metadata/query-values
                       (:db_id table)
-                      {:source-table (:id table)})))
+                      (merge (extract-query-opts opts)
```

Comment:
While the current solution is certainly unrefined to say the least, I'm not sure we want to push cost into data fetching because that locks us into only looking at cost at warehouse level, not computational cost (or we'll have duplication).